cloudlift/deployment/service_template_generator.py

```diff
@@ -65,7 +74,26 @@ def generate_service(self):
         self._fetch_current_desired_count()
         self._add_ecs_service_iam_role()
         self._add_cluster_services()
-        return to_yaml(self.template.to_json())
+        
+        letters = string.ascii_lowercase
+        key = ''.join(random.choice(letters) for i in range(20)) + '.yml'
+        if len(to_yaml(self.template.to_json())) > 51000:
+            try:
+                self.client.put_object(
+                    Body=to_yaml(self.template.to_json()),
+                    Bucket=self.bucket_name,
+                    Key=key,
+                )
+                TemplateURL = 'https://' + self.bucket_name + '.s3.amazonaws.com/'+ key
+                return TemplateURL , 'TemplateURL', key
+            except ClientError as boto_client_error:
+                error_code = boto_client_error.response['Error']['Code']
+                if error_code == 'AccessDenied':
+                    raise UnrecoverableException("Access Denied")
```

Comment:
Error should provide more details. Return the bucket name to which permissions are denied. "Unable to store cloudlift service template in S3 at <s3 bucket>"